Add Collective Name to the Missing Receipt Email

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/5949

znarf commented on templates/emails/collective.expense.missing.receipt.hbs:
```diff
@@ -11,7 +11,7 @@ Subject: Action Required: Submit receipt for card expense
 {{#if user}}
 A card attached to {{fromCollective.name}} was charged {{currency expense.amount currency=expense.currency}} and we need the responsible admin to input additional
-information.
+information. This expense was submitted to {{collective.name}}.
```
Why do we only have this under the `#if user` condition?

znarf replied:
Looks like this is a remain that was made useless after this refactor https://github.com/opencollective/opencollective-api/pull/6874  and can be removed if we're not using it anymore.

SudharakaP replied:
Removed the else part. 👍🏽

Betree commented on templates/emails/collective.expense.missing.receipt.hbs:
```diff
@@ -11,7 +11,7 @@ Subject: Action Required: Submit receipt for card expense
 {{#if user}}
 A card attached to {{fromCollective.name}} was charged {{currency expense.amount currency=expense.currency}} and we need the responsible admin to input additional
-information.
+information. This expense was submitted to {{collective.name}}.
```
Would be great for this to be a link, and same for `fromCollective`

SudharakaP replied:
Done. 👍🏽

RichardLitt commented:
Ah, shoot. I edited and submitted a PR for https://github.com/opencollective/opencollective-api/pull/7936 before I saw this edit. I added data to make compiling easier, and changed the language a bit - maybe we could take some edits from that, @SudharakaP?

SudharakaP commented:
> Ah, shoot. I edited and submitted a PR for #7936 before I saw this edit. I added data to make compiling easier, and changed the language a bit - maybe we could take some edits from that, @SudharakaP?

I've incorporated changes from that PR here as well and removed the unnecessary condition as pointed out at, https://github.com/opencollective/opencollective-api/pull/7934#discussion_r969219392. The new template looks like,

![image](https://user-images.githubusercontent.com/12435965/189982732-73f2da8b-e77b-4a3f-8010-beb2fd01647f.png)

RichardLitt commented:
The only thing I wonder is if there is a way we could mark the title and collective name with quotes. I think "Team dinner card expense" is an awkward phrase to parse. Maybe: `Please submit the receipt for the "Team Dinner" card expense` would be more natural?

SudharakaP commented:
> The only thing I wonder is if there is a way we could mark the title and collective name with quotes. I think "Team dinner card expense" is an awkward phrase to parse. Maybe: `Please submit the receipt for the "Team Dinner" card expense` would be more natural?

Done; so it looks;

![image](https://user-images.githubusercontent.com/12435965/190115108-cbcab5d3-73a1-4ee1-ba31-a461927a0a0d.png)

RichardLitt commented:
Thanks. Looks good to me.

znarf commented on templates/emails/collective.expense.missing.receipt.hbs:
```diff
@@ -5,16 +5,12 @@ Subject: Action Required: Submit receipt for card expense
-{{else}}
-You need to provide additional info about the card charge of {{currency expense.amount currency=expense.currency}} for {{fromCollective.name}}.
-{{/if}}
+A card attached to <a href="{{config.host.website}}/{{collective.slug}}">{{collective.name}}</a> was charged {{currency expense.amount currency=expense.currency}} for {{expense.description}} and we need the responsible admin to input additional information.
```
What if it's a project or an event ? The slug will not be enough.

znarf replied:
Why have we added expense.description ?

znarf replied:
Expense description is NOT looking like "Team Dinner" in production so it's super misleading to take this as an example.

The current expense description looks like this: "Virtual Card charge: Zoom". Quite different ...

Betree replied:
The case for events/projects URLs is difficult to handle in emails, we usually don't have access to the parents in there; I voluntarily skipped it too in https://github.com/opencollective/opencollective-api/pull/7937. The redirect should do the job, we could iterate in the future by auto-loading parents from collectives before generating templates.

SudharakaP replied:
@Betree : I propose a fix here, https://github.com/opencollective/opencollective-api/pull/7949 using the `getUrlPath` function. Let me know if that makes things better? 🤔 

@znarf : I've replaced the description with `fromCollective.name`.
